app/controllers/openid_connect/authorization_controller.rb

```diff
@@ -23,9 +23,13 @@ def index
 
     def create
       result = @authorize_form.submit(current_user, session.id)
-      analytics.track_event(Analytics::OPENID_CONNECT_ALLOW, result.except(:redirect_uri))
+      analytics_attributes = result.to_h
 
-      if (redirect_uri = result[:redirect_uri])
+      analytics.track_event(
+        Analytics::OPENID_CONNECT_ALLOW, analytics_attributes.except(:redirect_uri)
+      )
+
+      if (redirect_uri = analytics_attributes[:redirect_uri])
```

Comment:
`redirect_uri` should not be a key of a variable named `analytics_attributes` IMO, what about something more like:

```
analytics_attributes = result.to_h.except(:redirect_uri)

...
if (redirect_uri = result.extra[:redirect_uri]) 
```

Or maybe we should just pull `redirect_uri` out of the `result` hash and make it a separate method?